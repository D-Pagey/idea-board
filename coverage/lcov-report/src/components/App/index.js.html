<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/App/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/App</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">35.29% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/34</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
&nbsp;
import {
  AppWrapper,
  Title,
  AddIcon,
  SortAlpha,
  SortDate,
  SortDiv,
} from './styles';
import initialState from './initialState';
import Board from '../Board';
&nbsp;
export default class App extends Component {
  state = {
    tiles: [],
  }
&nbsp;
  componentDidMount() {
    this.checkLocalStorage();
  }
&nbsp;
  checkLocalStorage = () =&gt; {
    const oldTiles = localStorage.getItem('tiles');
    let newTiles = [];
&nbsp;
    <span class="missing-if-branch" title="if path not taken" >I</span>if (localStorage.length &gt; 0) {
<span class="cstat-no" title="statement not covered" >      newTiles = JSON.parse(oldTiles);</span>
    } else {
      newTiles = initialState;
    }
&nbsp;
    return this.setState({ tiles: newTiles }, this.saveStateToLocalStorage);
  }
&nbsp;
  saveStateToLocalStorage = () =&gt; {
    const { tiles } = this.state;
&nbsp;
    <span class="missing-if-branch" title="else path not taken" >E</span>if (tiles.length &gt; 0) {
      localStorage.setItem('tiles', JSON.stringify(tiles));
    } else {
<span class="cstat-no" title="statement not covered" >      localStorage.removeItem('tiles');</span>
    }
  }
&nbsp;
  sortDate = <span class="fstat-no" title="function not covered" >(a</span>rray) =&gt; {
    const sortedArray = <span class="cstat-no" title="statement not covered" >[...array].sort(<span class="fstat-no" title="function not covered" >(a</span>, b) =&gt; <span class="cstat-no" title="statement not covered" >b.date &gt; a.date)</span>;</span>
<span class="cstat-no" title="statement not covered" >    this.setState({ tiles: sortedArray }, this.saveStateToLocalStorage);</span>
  }
&nbsp;
  sortAlpha = <span class="fstat-no" title="function not covered" >(a</span>rray) =&gt; {
    const sortedArray = <span class="cstat-no" title="statement not covered" >[...array]</span>
      .sort(<span class="fstat-no" title="function not covered" >(a</span>, b) =&gt; <span class="cstat-no" title="statement not covered" >a.title.toLowerCase() &gt; b.title.toLowerCase())</span>;
<span class="cstat-no" title="statement not covered" >    this.setState({ tiles: sortedArray }, this.saveStateToLocalStorage);</span>
  }
&nbsp;
  handleChange = <span class="fstat-no" title="function not covered" >(i</span>ndex, event) =&gt; {
    const { tiles } = <span class="cstat-no" title="statement not covered" >this.state;</span>
    const newTiles = <span class="cstat-no" title="statement not covered" >[...tiles];</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    newTiles[index][event.target.name] = event.target.value;</span>
<span class="cstat-no" title="statement not covered" >    newTiles[index].updated = new Date().toLocaleTimeString().slice(0, 5);</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.setState({ tiles: newTiles }, this.saveStateToLocalStorage);</span>
  }
&nbsp;
  deleteTile = <span class="fstat-no" title="function not covered" >(i</span>d) =&gt; {
    const { tiles } = <span class="cstat-no" title="statement not covered" >this.state;</span>
&nbsp;
    const updatedTiles = <span class="cstat-no" title="statement not covered" >tiles.filter(<span class="fstat-no" title="function not covered" >el</span>ement =&gt; <span class="cstat-no" title="statement not covered" >element.id !== id)</span>;</span>
<span class="cstat-no" title="statement not covered" >    this.setState({ tiles: updatedTiles }, this.saveStateToLocalStorage);</span>
  }
&nbsp;
  addTile = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const { tiles } = <span class="cstat-no" title="statement not covered" >this.state;</span>
    const { length } = <span class="cstat-no" title="statement not covered" >tiles;</span>
&nbsp;
    const newTile = <span class="cstat-no" title="statement not covered" >{</span>
      title: `Genius idea #${length + 1}`,
      description: '',
      date: new Date().toLocaleTimeString().slice(0, 5),
      id: Math.random(),
    };
&nbsp;
    const moreTiles = <span class="cstat-no" title="statement not covered" >[...tiles, newTile];</span>
<span class="cstat-no" title="statement not covered" >    this.setState({ tiles: moreTiles }, this.saveStateToLocalStorage);</span>
  }
&nbsp;
  render() {
    const { tiles } = this.state;
&nbsp;
    const content = tiles.length &gt; 0 &amp;&amp; (
      &lt;Board
        tiles={tiles}
        deleteTile={this.deleteTile}
        handleChange={this.handleChange}
      /&gt;
    );
&nbsp;
    return (
      &lt;AppWrapper&gt;
        &lt;Title&gt;Idea Board&lt;/Title&gt;
        &lt;SortDiv&gt;
          &lt;SortAlpha onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.sortAlpha(tiles)}</span>&gt;sort_by_alpha&lt;/SortAlpha&gt;
          &lt;SortDate onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.sortDate(tiles)}</span>&gt;access_time&lt;/SortDate&gt;
        &lt;/SortDiv&gt;
        {content}
        &lt;AddIcon onClick={this.addTile}&gt;add_circle_outline&lt;/AddIcon&gt;
      &lt;/AppWrapper&gt;
    );
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 20 2019 00:52:51 GMT+0000 (GMT Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
