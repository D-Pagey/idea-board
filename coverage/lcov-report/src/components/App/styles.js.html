<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/App/styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/App</a> styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from 'styled-components';
&nbsp;
export const AppWrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 5px;
`;
&nbsp;
export const Title = styled.h1`
  font-size: 20px;
  margin: 20px 0 10px;
  text-align: center;
`;
&nbsp;
export const AddIcon = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  cursor: pointer;
  font-size: 30px;
  margin-top: 5px;
`;
&nbsp;
export const SortDiv = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 5px;
  width: 4.5rem;
`;
&nbsp;
export const SortAlpha = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  cursor: pointer;
  font-size: 30px;
`;
&nbsp;
export const SortDate = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  cursor: pointer;
  font-size: 30px;
`;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 20 2019 00:52:51 GMT+0000 (GMT Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
