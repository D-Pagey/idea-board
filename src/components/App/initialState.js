export default [{
  title: 'Hello',
  description: 'This is a description',
  date: new Date().toLocaleTimeString().slice(0, 5),
  id: 0,
},
{
  title: '2nd tile',
  description: 'I love RB6',
  date: new Date().toLocaleTimeString().slice(0, 5),
  id: 1,
}];
