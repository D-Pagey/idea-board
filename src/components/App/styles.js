import styled from 'styled-components';

export const AppWrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 5px;
`;

export const Title = styled.h1`
font-size: 20px;
margin: 20px 0 10px;
text-align: center;
`;

export const AddIcon = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  cursor: pointer;
  font-size: 30px;
  margin-top: 5px;
`;

export const SortDiv = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 5px;
  width: 4.5rem;
`;

export const SortAlpha = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  cursor: pointer;
  font-size: 30px;
`;

export const SortDate = styled.i.attrs({
  className: 'material-icons',
  role: 'button',
  tabIndex: '0',
})`
  cursor: pointer;
  font-size: 30px;
`;
