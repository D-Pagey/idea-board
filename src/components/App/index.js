import React, { Component } from 'react';

import Board from '../Board';
import { STORAGE_KEY } from './constants';
import initialState from './initialState';
import {
  AppWrapper,
  Title,
  AddIcon,
  SortAlpha,
  SortDate,
  SortDiv,
} from './styles';

export default class App extends Component {
  state = {
    tiles: initialState,
  }

  componentDidMount() {
    this.checkLocalStorage();
  }

  checkLocalStorage = () => {
    const storedTiles = localStorage.getItem(STORAGE_KEY);
    if (storedTiles) this.setState({ tiles: JSON.parse(storedTiles) });
  }

  saveStateToLocalStorage = () => {
    const { tiles } = this.state;
    localStorage.setItem(STORAGE_KEY, JSON.stringify(tiles));
  }

  sortDate = (array) => {
    const sortedArray = [...array].sort((a, b) => b.date > a.date);
    this.setState({ tiles: sortedArray }, this.saveStateToLocalStorage);
  }

  sortAlpha = (array) => {
    const sortedArray = [...array]
      .sort((a, b) => a.title.toLowerCase() > b.title.toLowerCase());
    this.setState({ tiles: sortedArray }, this.saveStateToLocalStorage);
  }

  handleChange = (index, event) => {
    const { tiles } = this.state;
    const newTiles = [...tiles];

    newTiles[index][event.target.name] = event.target.value;
    newTiles[index].updated = new Date().toLocaleTimeString().slice(0, 5);

    this.setState({ tiles: newTiles }, this.saveStateToLocalStorage);
  }

  deleteTile = (id) => {
    const { tiles } = this.state;
    const updatedTiles = tiles.filter(element => element.id !== id);

    this.setState({ tiles: updatedTiles }, this.saveStateToLocalStorage);
  }

  addTile = () => {
    const { tiles } = this.state;
    const moreTiles = [...tiles, {
      title: `Genius idea #${tiles.length + 1}`,
      description: '',
      date: new Date().toLocaleTimeString().slice(0, 5),
      id: Math.random(),
    }];

    this.setState({ tiles: moreTiles }, this.saveStateToLocalStorage);
  }

  render() {
    const { tiles } = this.state;

    return (
      <AppWrapper>
        <Title>Idea Board</Title>

        <SortDiv>
          <SortAlpha onClick={() => this.sortAlpha(tiles)}>sort_by_alpha</SortAlpha>
          <SortDate onClick={() => this.sortDate(tiles)}>access_time</SortDate>
        </SortDiv>

        {tiles.length > 0 && (
          <Board
            tiles={tiles}
            deleteTile={this.deleteTile}
            handleChange={this.handleChange}
          />
        )}

        <AddIcon onClick={this.addTile}>add_circle_outline</AddIcon>
      </AppWrapper>
    );
  }
}
