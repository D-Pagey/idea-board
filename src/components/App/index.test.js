import React from 'react';
import { shallow } from 'enzyme';

import { STORAGE_KEY } from './constants';
import App from '.';

describe('App component', () => {
  it('should render', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should check local storage on mount for tiles', () => {
    const wrapper = shallow(<App />);
    const spy = jest.spyOn(wrapper.instance(), 'checkLocalStorage');
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should render tiles from local storage', () => {
    localStorage.getItem.mockImplementationOnce(() => '[]');
    const wrapper = shallow(<App />);
    expect(localStorage.getItem).toHaveBeenCalledWith(STORAGE_KEY);
    expect(wrapper.find(<Board />)).toHaveLength(1);
  });
});

/** Things to test:-
 * CDM - calls checkLocalStorage
 * checkLocalStorage, test it setState with some tiles
 * checkLocalStorage, test it call this.saveStateTo.....
 * saveStateToLocalStorag, test it saves to localStorage
 * sortdate = test it works as intended
 * sortalpha = test it works as intended
 * handleChange = test it works as intended
 * deleteTile = test it works as intended
 * addTile = test it works as intended
 * SortAlhpa = test the onclick
 * SortDate = test the onclick
 * addIcon = test the onclick
 */
