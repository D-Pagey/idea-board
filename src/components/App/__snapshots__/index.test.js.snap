// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render 1`] = `
<styled.div>
  <styled.h1>
    Idea Board
  </styled.h1>
  <styled.div>
    <styled.i
      onClick={[Function]}
    >
      sort_by_alpha
    </styled.i>
    <styled.i
      onClick={[Function]}
    >
      access_time
    </styled.i>
  </styled.div>
  <Board
    deleteTile={[Function]}
    handleChange={[Function]}
    tiles={
      Array [
        Object {
          "date": "01:43",
          "description": "This is a description",
          "id": 0,
          "title": "Hello",
        },
        Object {
          "date": "01:43",
          "description": "I love RB6",
          "id": 1,
          "title": "2nd tile",
        },
      ]
    }
  />
  <styled.i
    onClick={[Function]}
  >
    add_circle_outline
  </styled.i>
</styled.div>
`;
